lib/terraform/runner.rb

```diff
@@ -0,0 +1,293 @@
+      # @return [Terraform::Runner::Response] Response object with final result of terraform run
+      def run(input_vars, template_path, tags: nil, credentials: [], env_vars: {})
+        _log.debug("Run template: #{template_path}")
+        create_stack_job_and_wait_until_completes(
+          template_path,
+          :input_vars  => input_vars,
+          :tags        => tags,
+          :credentials => credentials,
+          :env_vars    => env_vars
+        )
+      end
+
+      # Fetch terraform-runner job result/status by stack_id
+      #
+      # @param stack_id [String] stack_id from the terraforn-runner job
+      #
+      # @return [Terraform::Runner::Response] Response object with result of terraform run
+      def fetch_result_by_stack_id(stack_id)
+        retrieve_stack_job(stack_id)
+      end
+
+      # =================================================
+      # TerraformRunner Stack-API interaction methods
+      # =================================================
+      private
+
+      def server_url
+        ENV['TERRAFORM_RUNNER_URL'] || 'https://opentofu-runner:27000'
+      end
+
+      def server_token
+        ENV.fetch('TERRAFORM_RUNNER_TOKEN', nil)
+      end
+
+      def stack_job_interval_in_secs
+        ENV['TERRAFORM_RUNNER_STACK_JOB_CHECK_INTERVAL'].to_i
+      rescue
+        10
+      end
+
+      def stack_job_max_time_in_secs
+        ENV['TERRAFORM_RUNNER_STACK_JOB_MAX_TIME'].to_i
+      rescue
+        120
+      end
+
+      # Create to paramaters as used by terraform-runner api
+      #
+      # @param vars [Hash] Hash with key/value pairs that will be passed as input variables to the
+      #        terraform-runner run
+      # @return [Array] Array of {:name,:value}
+      def convert_to_cam_parameters(vars)
+        parameters = []
+        vars&.each do |key, value|
+          parameters.push(
+            {
+              :name  => key,
+              :value => value
+            }
+          )
+        end
+        parameters
+      end
+
+      # create http client for terraform-runner rest-api
+      def terraform_runner_client
+        # TODO: verify ssl
+        verify_ssl = false
+
+        RestClient::Resource.new(
+          server_url,
+          :headers    => {:authorization => "Bearer #{server_token}"},
+          :verify_ssl => verify_ssl
+        )
+      end
+
+      def stack_tenant_id
+        # TODO: fix hardcoded tenant_id
+        'c158d710-d91c-11ed-9fee-d93323035b4e'
+      end
+
+      # Create TerraformRunner Stack Job
+      def create_stack_job(
+        template_path,
+        input_vars: [],
+        tags: nil,
+        credentials: [],
+        env_vars: {},
+        name: "stack-#{rand(36**8).to_s(36)}"
+      )
+        _log.info("start stack_job for template: #{template_path}")
+        tenant_id = stack_tenant_id
+
+        # Temp Zip File
+        # zip_file_path = Tempfile.new(%w/tmp .zip/).path
+        zip_file_path = Tempfile.new(%w[tmp .zip]).path
```

Comment:
If you use `Tempfile.create` with a block you don't need to explicitly unlink the temp file, https://ruby-doc.org/stdlib-3.1.1/libdoc/tempfile/rdoc/Tempfile.html#method-c-create

```
Tempfile.create(%w[opentofu-runner-payload .zip]) do |zip_file|
  create_zip_file_from_directory(zip_file.path, template_path)
end
```

And then you don't need the `ensure` https://github.com/ManageIQ/manageiq-providers-embedded_terraform/pull/7/files#diff-6c77e1aede2fb01f9a96887c7b650c186b8daad450c61e621c389b84ffcd1d21R175

Reply:
done